include/RAJA/policy/cuda/policy.hpp

```diff
@@ -190,9 +289,44 @@ namespace policy
 namespace cuda
 {
 
+struct DeviceConstants
+{
+  RAJA::Index_type WARP_SIZE;
+  RAJA::Index_type MAX_BLOCK_SIZE;
+  RAJA::Index_type MAX_WARPS;
+  RAJA::Index_type ATOMIC_DESTRUCTIVE_INTERFERENCE_SIZE; // basically the cache line size of the cache level that handles atomics
+  RAJA::Index_type ATOMIC_MAX_CONCURRENT_SIZE;
+  RAJA::Index_type ATOMIC_MAX_CONCURRENT_CACHE_LINES;
+
+  constexpr DeviceConstants(RAJA::Index_type warp_size,
+                            RAJA::Index_type max_block_size,
+                            RAJA::Index_type atomic_cache_line_bytes,
+                            RAJA::Index_type atomic_max_concurrency_bytes) noexcept
+    : WARP_SIZE(warp_size)
+    , MAX_BLOCK_SIZE(max_block_size)
+    , MAX_WARPS(max_block_size / warp_size)
+    , ATOMIC_DESTRUCTIVE_INTERFERENCE_SIZE(atomic_cache_line_bytes)
+    , ATOMIC_MAX_CONCURRENT_SIZE(atomic_max_concurrency_bytes)
+    , ATOMIC_MAX_CONCURRENT_CACHE_LINES(atomic_max_concurrency_bytes / atomic_cache_line_bytes)
+  { }
+};
+
+//
+// Operations in the included files are parametrized using the following
+// values for CUDA warp size and max block size.
+//
+// constexpr DeviceConstants device_constants(32, 1024, 32, 65'536); // V100
```

Comment:
Can this commented out line be removed?

Reply:
This is part of the performance tuning that I'm still working on.

Reply:
gone now